src/dev-app/theme-m3.scss

```diff
@@ -2,6 +2,22 @@
 @use '@angular/material' as mat;
 @use '@angular/material-experimental' as matx;
 
+// TODO(mmalerba): Consider adding this as a back-compat API for users who want it, rather than just
+//  a demo thing.
+@mixin color-variants-back-compat($theme) {
+  .mat-primary {
+    &.mat-icon { @include mat.icon-color($theme, $color-variant: primary); }
```

Comment:
Would be interesting to see how we can provide this for users (if we think they would want it). Maybe each theme file provides some back-compat mixin?

Reply:
I think we could do component-by-component mixins, or we could just do a single back-compat mixin like I have in this demo. Since its just a back-compat thing anyways, I kind of like the idea of just having a single combined one so its simple for people

Reply:
Yeah I don't feel like it's something I really want to encourage folks to do, but instead be a reference for them, so keeping it a single mixin makes sense. Unfortunately, It'll add a lot of a code size for someone to use since it's including everything. 

I wonder how this would fit into the story of helping folks migrate, especially internally. Maybe we ought to figure out what that looks like and it might determine what we do here. 

That said, it doesn't need to block this PR - just something we should consider until this is released

Reply:
Yeah, FWIW this is code people already have in their apps for M2, so this isn't adding _additional_ CSS, but putting it all in one mixin does mean they can't be as aggressive about removing component-by-component it once its no longer needed.